velox/functions/prestosql/Sequence.cpp

```diff
@@ -40,8 +41,8 @@ int64_t toInt64(Timestamp value) {
   return value.toMillis();
 }
 
-template <typename T>
-T add(T value, int64_t steps);
+template <typename T, typename K>
```

Comment:
Since we only have `int32_t` and `int64_t` for steps and there is no array of `K`, we don't need a type parameter to duplicate the generated code for this.  Just cast int32 to int64 should be ok.

Reply:
But then we will not be able to distinguish IntervalDayToSecond and IntervalYearToMonth if we make them both int64_t in `add` function 

```
template <>
Date add(Date value, int64_t steps) {
  return Date(value.days() + steps);
}

template <>
Date add(Date value, int32_t steps) {
  return addToDate(value, DateTimeUnit::kMonth, steps);
}
```

Reply:
We can change `typename K` to `bool kIsStepUnitMonth`, this better conveys the purpose.

Reply:
It would add several `if else` and is a bit overkill, just added the comment to make it more self-explanatory

Reply:
You can do it like this:
```c++
using Months = int32_t;
using Days = int64_t;

template <>
Date add(Date value, Days steps) {
  return Date(value.days() + steps);
}

template <>
Date add(Date value, Months steps) {
  return addToDate(value, DateTimeUnit::kMonth, steps);
}
```

Reply:
Didn't add `Days` since for some other functions, it doesn't mean days, eg:
```
template <>
Timestamp add(Timestamp value, int64_t steps) {
  return Timestamp::fromMillis(value.toMillis() + steps);
}
```

Added `Months` and removed the comment, thanks!

Reply:
You can leave `int64_t` where it makes sense and only replace it for `add(Date value, Days steps)`